xbmc/video/VideoDatabase.cpp

```diff
@@ -3034,28 +3046,89 @@ bool CVideoDatabase::SetFileForEpisode(const std::string& fileAndPath, int idEpi
   return false;
 }
 
-bool CVideoDatabase::SetFileForMovie(const std::string& fileAndPath, int idMovie, int idFile)
+bool CVideoDatabase::SetFileForMovie(const std::string& fileAndPath, int idMovie)
 {
+  const int idFile{AddFile(fileAndPath)};
+  if (idFile < 0)
+    return false;
+
+  const int oldIdFile{GetFileIdByMovie(idMovie)};
+  if (oldIdFile < 0)
+    return false;
+
   try
   {
     assert(m_pDB->in_transaction());
+    std::string sql = PrepareSQL("UPDATE movie SET idFile=%i WHERE idFile=%i AND idMovie=%i",
+                                 idFile, oldIdFile, idMovie);
+    m_pDS->exec(sql);
 
-    std::string sql = PrepareSQL("UPDATE movie SET c22='%s', idFile=%i WHERE idMovie=%i",
-                                 fileAndPath.c_str(), idFile, idMovie);
+    sql = PrepareSQL(
+        "UPDATE videoversion SET idFile=%i WHERE idFile=%i AND media_type='movie' AND idMedia=%i",
+        idFile, oldIdFile, idMovie);
     m_pDS->exec(sql);
-    sql = PrepareSQL("UPDATE videoversion SET idFile=%i WHERE idMedia=%i AND media_type='movie'",
-                     idFile, idMovie);
+
+    sql = PrepareSQL("UPDATE art SET media_id=%i WHERE media_id=%i AND media_type='videoversion'",
+                     idFile, oldIdFile);
     m_pDS->exec(sql);
 
+    if (GetSingleValueInt(PrepareSQL("SELECT idFile FROM streamdetails WHERE idFile = %i", idFile),
+                          m_pDS) == 0)
```

Comment:
In this case there can't be a record with idFile == 0 in the table so nothing bad can happen but that's a dangerous way to test record existence.
"SELECT 1 FROM ..." would be guaranteed to work regardless of field types or values.

That said, GetSingleValueInt is not well suited to this use case imo and a classic DS->query followed by if (eof) is clearer, or a new Exist() helper would be better. A couple other spots in this commit use the same pattern. There may be other existing places as well.

Taking a step back, maybe the call to SetFile should happen before the stream details update of SaveFilteState. That way, existing streamdetails can be updated unconditionnally.